src/generic/stage2/numberparsing.h

```diff
@@ -497,22 +497,30 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
       }
       // we over-decrement by one when there is a '.'
       digit_count -= int(start - start_digits);
-      if (digit_count >= 19) {
+      if (unlikely(digit_count >= 19)) {
```

Comment:
I challenge this change. The whole code section is already marked as unlikely, and you have now marked something as unlikely within an unlikely region. Either this brings no benefit whatsoever, in which case it should be removed, or else, if it brings benefits, then we have to question the whole code structure. My rule of thumb is that you should never have a chain of unlikely. Either a code region is unlikely or not. It should not be meaningful to be unlikely inside an unlikely region. Now, this is not a hard rule, but you are violating my rule of thumb.

Reply:
I agree. Also, this statement is not unlikely, given that we only get here if there were 19 digits to start with.

Reply:
(I think adding this was a mistake on my part, I probably thought it was the earlier if() and was puzzled why I would have removed it, so added it back.)